docs/faq.md

```diff
@@ -1,21 +1,179 @@
+
+Yes, as Fn packages and deploys all functions as Docker containers it’s possible to provide a custom Docker image that includes your function.  As long as your container implements the Fn function contract it can contain anything.
+
+### What is the lifecycle of a function?
+
+Functions are packaged as Docker images and by default individual containers are created to handle a function request and are then destroyed.  However, [Hot Functions](hot-functions.md) are not disposed of after handling a single request.
+
+### What’s a `Hot Function`?
+
+`Hot Functions` are functions that are not destroyed after a single use but are retained and used to handle subsequent requests.  Hot Functions must accept HTTP input.
+
+<a id="General"></a>
+## General
+
+### What is the Fn Project’s license?
+
+Apache 2.0
+
+### What is the URL of the project website?
+
+<http://fnproject.io>
+
+### Where is the source code hosted?
+
+All of the project code is on Github at <http://github.com/fnproject>.
+
+### Is Fn an entirely new platform?
+
+No, Fn is derived from the well-received IronFunctions project.  The core IronFunctions team are now at Oracle so Fn is simply the next evolutionary step of IronFunctions built by the original developers.
+
+### How do you pronounce “Fn”?
+
+The project name is pronounced “F" "N”.
+
+### Do you provide any UI/tool to track our Fn status?
+
+Fn has a dashboard that can be found at <https://github.com/fnproject/ui>. Fn Flow also has an experimental dashboard at <https://github.com/fnproject/flowui>.
+
+### What is Fn Flow?
+
+Fn Flow is a [Java API](https://github.com/fnproject/fn-java-fdk/blob/master/docs/FnFlowsUserGuide.md) and [corresponding service](https://github.com/fnproject/completer) that helps you create complex, long-running, fault-tolerant functions using a promises-style asynchronous API. Check out the [Fn Flow docs](https://github.com/fnproject/fn-java-fdk/blob/master/docs/FnFlowsUserGuide.md) for more information.
+
+
+<a id="Network"></a>
+## Network
+
+### Can Fn do IO over HTTP or in general any TCP based communication?
+
+Like most functions platforms we support HTTP.
+
+### Can an Fn function connect to various endpoints to read/write data?
+
+Yes, a function is not restricted in what it can connect to.
+
+### Any limits on the network bandwidth?
+
+We currently don't yet offer Fn as a managed service which would manage network traffic.  Currently, there are no restrictions in the core Fn platform.
+
+<a id="Performance"></a>
+## Performance
+
+### How long can a function run? Can we make it run forever?
+
+We support 'hot functions' (see the end of [this tutorial](https://github.com/fnproject/tutorials/blob/master/JavaFDKIntroduction/README.md) for an example).  Hot functions will continue to live if they are used but, if not, will eventually be cleaned up.
+
+Function timeout is configurable. Please see <https://github.com/fnproject/fn/blob/master/docs/function-file.md>
```

Comment:
but it has upper limits, i.e. you can't run sync function longer than 120 seconds.

Reply:
Updated by incorporating your comment.